regolith/tools.py

```diff
@@ -1322,3 +1326,50 @@ def is_fully_appointed(person, begin_date, end_date):
             else:
                 print("appointment gap for {} on {}".format(person.get('_id'), str(day)))
     return status
+
+def key_value_pair_filter(collection, arguments, keys=None):
+    """Retrieves a list of all documents from the collection where the fragment
+        appears in any one of the given fields
+
+        Parameters
+        ----------
+        collection: generator
+            The collection containing the documents
+        arguments: list
+            The name of the fields to look for and their accompanying substring
+        keys: list
+            The name of the fields to return from the search
+
+        Returns
+        -------
+        str:
```

Comment:
this is wrong.  per your docstring, shouldn't it return a list of dicts (a filtered collection)? (maybe related to my comment above?)

It is probably worth spending a moment to think about this.  How do we want this to behave and then how to do the code design.  It might be better to just extend fragment retrieval to take lists, but also, maybe not if we are making this into a kind of decorator.